### PR TITLE
Fix corner case: nesting all columns of an empty data frame

### PR DESCRIPTION
**Pull request by krlmlr**
Addresses part of #734.

### CODE REVIEW

File: tests/testthat/test-nest.R [hunk 1]
```diff
@@ -68,7 +68,7 @@ test_that("nesting works for empty data frames", {
 
   out <- nest(df, data = c(x, y))
   expect_named(out, "data")
-  expect_equal(nrow(out), 0L)
+  expect_equal(nrow(out), 1L)
```
Comment (hadley) [inline tests/testthat/test-nest.R:71]: I don't understand why this yields 1 row but the example yields 0 rows? 

I'm no longer confident that the `summarise()` justification is quite right (because `nest()` isn't necessarily perfectly aligned with dplyr reasoning), so unless you feel strongly about this, I'd rather close, preferring to fix related issues in `nest_by()`.

**Comment by krlmlr** [comment]
I'd rather reimplement using `nest_by()` .
